### ISSUE TITLE
Property Link Factory Memory Leak

### ISSUE THREAD

**Issue by RobertBColton** [IsmAvatar/LateralGM#531]
So 49115ddf5235718a31443ba90e573e8a5da425f5 caused unnecessary retention of the resource frames because the property link factory class is statically holding a strong reference to the factories of an owner which are holding strong references to their owner through the exception listener. That is a circular reference which causes the factories in the owned map to always be reachable. We can verify this by printing a message in resource frame's finalize. Also need to mention there is in fact what is arguably a bug in the Windows look and feel (probably the desktop manager) that causes the last closed MDI frame to not be garbage collected, and this issue is talked about on the internet.

I don't see this a big deal because at least people can edit properties without an exception after reopening the frames now. However, we should address this and there's a couple of ways we can do it.
1) Move the owner handling of the property link factories directly into the resource frame, and possibly require creating a factory through the base resource frame.
2) Change the factory owner map to use a second `WeakHashMap` for the value type instead of `List` so the cycle is broken.
3) Change the factory to have only a `WeakReference` to its owner. This is my least favorite because the class still always holds strong static references to the factories, which I don't think is a good idea.
4) Change the factory owner map to use the `WeakArrayList` class we have. There's issues with this though because that class's implementation has inconsistencies with the official `WeakHashMap` in the Java API. It directly extends an `ArrayList` with `WeakReference` as the value type, which requires direct use of WeakReferences in the list, which `WeakHashMap` does not.

**Comment by RobertBColton** [comment]
Closing as resolved by c85f9581f9266e74bb52c37218e353430d2d0191 which makes it the responsibility of the owner to manage its own property link factories. All of the other solutions to this suck because they are just adding unnecessary indirection to the problem. The real original issue here was just the mutate in notification caused by the fact that our property links don't handle the bidirectionality well. Some of the property links attempt to by checking if the value is actually different, but it's not quite the same as removing the listener during the synchronization. One other solution we could try later and I didn't mention is. For now, the memory leak is gone and the exception is gone so the user won't notice this anymore.

**Comment by RobertBColton** [comment]
Just to show what I mean that the issue is with DocumentLink still. If I go to game settings frame and bind two different text fields, with different documents, to the same map property, I can reproduce the original mutate in notification exception.
```java
		plf.make(author.getDocument(),PGameSettings.AUTHOR);
		plf.make(version.getDocument(),PGameSettings.AUTHOR);
```
